dipy/reconst/mcsd.py

```diff
@@ -0,0 +1,268 @@
+
+
+def _basic_delta(iso, m, n, theta, phi):
+    """Simple delta function
+    Parameters
+    ----------
+    iso: int
+        Number of tissue compartments for running the MSMT-CSD. Minimum
+        number of compartments required is 2.
+        Default: 2
+    m : int ``|m| <= n``
+        The order of the harmonic.
+    n : int ``>= 0``
+        The degree of the harmonic.
+    theta : array_like
+       inclination or polar angle
+    phi : array_like
+       azimuth angle
+    """
+    wm_d = shm.gen_dirac(m, n, theta, phi)
+    iso_d = [SH_CONST] * iso
+    return np.concatenate([iso_d, wm_d])
+
+
+class MultiShellDeconvModel(shm.SphHarmModel):
+    def __init__(self, gtab, response, reg_sphere=default_sphere, iso=2):
+        r"""
+        Multi-Shell Multi-Tissue Constrained Spherical Deconvolution
+        (MSMT-CSD) [1]_. This method extends the CSD model proposed in [2]_ by
+        the estimation of multiple response functions as a function of multiple
+        b-values and multiple tissue types.
+
+        Spherical deconvolution computes a fiber orientation distribution
+        (FOD), also called fiber ODF (fODF) [2]_. The fODF is derived from
+        different tissue types and thus overcomes the overestimation of WM in
+        GM and CSF areas.
+
+        The response function is based on the different tissue types
+        and is provided as input to the MultiShellDeconvModel.
+        It will be used as deconvolution kernel, as described in [2]_.
+
+        Parameters
+        ----------
+        gtab : GradientTable
+        response : tuple or AxSymShResponse object
+            A tuple with two elements. The first is the eigen-values as an (3,)
+            ndarray and the second is the signal value for the response
+            function without diffusion weighting.  This is to be able to
+            generate a single fiber synthetic signal. The response function
+            will be used as deconvolution kernel ([1]_)
+        reg_sphere : Sphere (optional)
+            sphere used to build the regularization B matrix.
+            Default: 'symmetric362'.
+        iso: int (optional)
+            Number of tissue compartments for running the MSMT-CSD. Minimum
+            number of compartments required is 2.
+            Default: 2
+
+        References
+        ----------
+        .. [1] Jeurissen, B., et al. NeuroImage 2014. Multi-tissue constrained
+               spherical deconvolution for improved analysisof multi-shell
+               diffusion MRI data
+        .. [2] Tournier, J.D., et al. NeuroImage 2007. Robust determination of
+               the fibre orientation distribution in diffusion MRI:
+               Non-negativity constrained super-resolved spherical
+               deconvolution
+        .. [3] Tournier, J.D, et al. Imaging Systems and Technology
+               2012. MRtrix: Diffusion Tractography in Crossing Fiber Regions
+        """
+        if not iso >= 2:
```

Comment:
Could you please add a test for this? That is, verify that the error is raised for number smaller than 2.